### DISCUSSION TITLE
2.0.0 beta release(s)

### DISCUSSION THREAD

**Discussion by jglev** [jglev/obsidian-apply-patterns-plugin#40] [Announcements]
The [`v2.0.0-beta0` release](https://github.com/jglev/obsidian-apply-patterns-plugin/releases/tag/2.0.0-beta0) is available for testing!

**Comment by jglev** [comment]
@FelipeRearden, if you would be willing to test any of these changes (if I may ask -- and if not, I'll understand!), I would be grateful for any feedback! This relates to a discussion we had in #9.

**Reply by FelipeRearden** [reply]
@jglev its an honor for me to participate in this beta test. I'm so happy !!!! 

Thank you very much for thinking about me for this new version. 

You can count on me 😃 !!!! 

Today is dinner/bed time for me but tomorrow first thing in the morning I will start my cycle of test 

Again, thank very much for considering me as tester for this plug-in that I love very much !!!!!!!

**Reply by FelipeRearden** [reply]
When you can, just let me know how we gonna organize my feedback (bugs, FR and ideas)

Should I use Discussions or Issues ?

**Reply by FelipeRearden** [reply]
Beta version v2.0.0-beta0 successfully installed !!!

Tomorrow I will start the my tests :)

**Reply by jglev** [reply]
Thank you very much! : )

> Should I use Discussions or Issues?

Let's use this existing Discussion as a central thread, and then split out separate Issues (rather than other Discussions) as we find things. Does that sound reasonable to you, too?

Again, thank you!

**Reply by FelipeRearden** [reply]
> Let's use this existing Discussion as a central thread, and then split out separate Issues (rather than other Discussions) as we find things. Does that sound reasonable to you, too?

It's perfect !!!

> Again, thank you!

My pleasure @jglev !!!! I'm so excited to take a look to the new features tomorrow 🙃

**Reply by FelipeRearden** [reply]
Hello @jglev !!!!!!

## Beta Test 01 -> v2.0.0-beta0

### Bugs
ZERO !!!!!!

### Test Environment 

Type of Patterns:
- Pattern with one rule
- Pattern with two rules

Commands 
- whole clipboard
- clipboard (line-by-line)

Without Commands -> applied direct from command palette 

Automation:
- Start from Command Palette 
- Start from hotkeys

Inputs
- single line
- Multiple lines
- Multiple cursors 

### Comments 
Everything worked fantastic. I have nothing else to say about the new features 

I just tested the new features 

I love the new inline validation while creating the rule. It's perfect 

In my humble opinion this release is ready for the public

The new features creates an amazing workflow to be merged with Carry-Forward. I'm gonna write on the Carry-Forward repository later to share with everybody 
- you can combine a pattern to change the `Default Link Text` to another text. This way I can create multiple types of Carry-Forward Link texts !!!!! 

After the public release, I will do the tests on iOS as well 🙏 

---

Let me know if you need any more tests !!!!

Have a great day!!!!!

**Reply by jglev** [reply]
@FelipeRearden, thank you very much! I appreciate your time! :rocket: And I look forward to reading your ideas going forward! : )

**Reply by FelipeRearden** [reply]
> @FelipeRearden, thank you very much! I appreciate your time! :rocket:

Thank you very for one more amazing release !!!! It was a pleasure for me to be a beata tester. 

Please I'm always available for more tests in the future 👀 

> And I look forward to reading your ideas going forward! : )

I will . Thank you very much for your patience :)
- I have two more for Carry-Forward (1 minor and 1 big)
- One for Apply-Patterns related to the UI 


I'm gonna test on iOS and I bring the result for you here. 

I wish you a fantastic day @jglev !!!!!!!!!!!
